TCC.Lib/TarCompressCrypt.cs

```diff
@@ -87,79 +72,88 @@ private static ParallelOptions ParallelOptions(string threads)
 
-            CleanupKey(block, option, key, keyCrypted, result, Mode.Compress);
+            CleanupKey(block, option, k, result, Mode.Compress);
 
             return result;
         }
 
-        private static void CleanupKey(Block block, TccOption option, string key, string keyCrypted, CommandResult result, Mode mode)
+        private static void CleanupKey(Block block, TccOption option, EncryptionKey key, CommandResult result, Mode mode)
         {
             if (option.PasswordMode == PasswordMode.PublicKey)
             {
                 // delete uncrypted pass
-                if (!String.IsNullOrEmpty(key))
+                if (!String.IsNullOrEmpty(key.KeyCrypted))
                 {
-                    File.Delete(Path.Combine(block.DestinationFolder, key));
+                    File.Delete(Path.Combine(block.DestinationFolder, key.Key));
                 }
                 // if error in compression, also delete encrypted passfile
-                if (mode == Mode.Compress && result.HasError && !String.IsNullOrEmpty(keyCrypted))
+                if (mode == Mode.Compress && result.HasError && !String.IsNullOrEmpty(key.KeyCrypted))
                 {
-                    File.Delete(Path.Combine(block.DestinationFolder, keyCrypted));
+                    File.Delete(Path.Combine(block.DestinationFolder, key.KeyCrypted));
                 }
             }
         }
 
-        private static void PrepareEncryptionKey(Block block, TccOption option, out string key, out string keyCrypted, CancellationToken cancellationToken)
+        public class EncryptionKey
+        {
+            public EncryptionKey(string key, string keyCrypted)
+            {
+                Key = key;
+                KeyCrypted = keyCrypted;
+            }
+
+            public string Key { get; }
+            public string KeyCrypted { get; }
+        }
+
+        private static async Task<EncryptionKey> PrepareEncryptionKey(Block block, TccOption option, CancellationToken cancellationToken)
         {
-            key = null;
-            keyCrypted = null;
+            string key = null;
+            string keyCrypted = null;
             if (option.PasswordMode == PasswordMode.PublicKey)
             {
                 key = block.ArchiveName + ".key";
                 keyCrypted = block.ArchiveName + ".key.encrypted";
                 // generate random passfile
-                var keyResult = GenerateRandomKey(key).Run(block.DestinationFolder, cancellationToken);
+                var keyResult = await GenerateRandomKey(key).Run(block.DestinationFolder, cancellationToken);
                 // crypt passfile
-                var keyCryptedResult = EncryptRandomKey(key, keyCrypted, option.PublicPrivateKeyFile).Run(block.DestinationFolder, cancellationToken);
+                var keyCryptedResult = await EncryptRandomKey(key, keyCrypted, option.PublicPrivateKeyFile).Run(block.DestinationFolder, cancellationToken);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused 'keyCryptedResult' local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1481)